src/main/capi/data_chunk-c.cpp

```diff
@@ -174,6 +174,38 @@ duckdb_vector duckdb_array_vector_get_child(duckdb_vector vector) {
 	return reinterpret_cast<duckdb_vector>(&duckdb::ArrayVector::GetEntry(*v));
 }
 
+duckdb_vector duckdb_map_vector_get_keys(duckdb_vector vector) {
+	if (!vector) {
+		return nullptr;
+	}
+	auto v = reinterpret_cast<duckdb::Vector *>(vector);
+	return reinterpret_cast<duckdb_vector>(&duckdb::MapVector::GetKeys(*v));
```

Comment:
Can we check if the vector has the right type before calling the vector-type specific methods?
Not doing that likely results in an internal exception